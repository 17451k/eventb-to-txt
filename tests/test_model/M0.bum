<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<org.eventb.core.machineFile org.eventb.core.configuration="org.eventb.core.fwd" version="5">
<org.eventb.core.event name="'" org.eventb.core.comment="comment" org.eventb.core.convergence="0" org.eventb.core.extended="false" org.eventb.core.label="INITIALISATION">
<org.eventb.core.action name="'" org.eventb.core.assignment="var1 ≔ 1" org.eventb.core.label="act1"/>
<org.eventb.core.action name="(" org.eventb.core.assignment="var2 ≔ {1}" org.eventb.core.label="act2"/>
</org.eventb.core.event>
<org.eventb.core.variable name="(" org.eventb.core.comment="comment" org.eventb.core.identifier="var1"/>
<org.eventb.core.variable name=")" org.eventb.core.identifier="var2"/>
<org.eventb.core.invariant name="*" org.eventb.core.comment="comment" org.eventb.core.label="inv1" org.eventb.core.predicate="var1 ∈ ℕ"/>
<org.eventb.core.invariant name="+" org.eventb.core.comment="&#10;" org.eventb.core.label="inv2" org.eventb.core.predicate="var2 ⊆ ℕ"/>
<org.eventb.core.invariant name="," org.eventb.core.comment="multiline&#10;comment" org.eventb.core.label="inv3" org.eventb.core.predicate="1 ∈ ℕ" org.eventb.core.theorem="true"/>
<org.eventb.core.event name="." org.eventb.core.convergence="1" org.eventb.core.extended="false" org.eventb.core.label="evt1">
<org.eventb.core.parameter name="'" org.eventb.core.comment="comment" org.eventb.core.identifier="prm1"/>
<org.eventb.core.guard name="(" org.eventb.core.label="grd1" org.eventb.core.predicate="prm1 = 1"/>
<org.eventb.core.action name=")" org.eventb.core.assignment="var1 ≔ var1 − prm1" org.eventb.core.comment="comment" org.eventb.core.label="act1"/>
<org.eventb.core.action name="*" org.eventb.core.assignment="var2 ≔ {1}" org.eventb.core.label="act2"/>
<org.eventb.core.guard name="-" org.eventb.core.comment="comment" org.eventb.core.label="grd2" org.eventb.core.predicate="var1 ≠ prm1"/>
<org.eventb.core.guard name="," org.eventb.core.label="grd3" org.eventb.core.predicate="var1 &gt; 0"/>
<org.eventb.core.guard name="." org.eventb.core.label="grd4" org.eventb.core.predicate="var1 ≠ 0" org.eventb.core.theorem="true"/>
</org.eventb.core.event>
<org.eventb.core.variant name="/" org.eventb.core.comment="comment" org.eventb.core.expression="var1"/>
</org.eventb.core.machineFile>
