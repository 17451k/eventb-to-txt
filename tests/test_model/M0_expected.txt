machine M0

variables
    var1 // comment
    var2

invariants
    @inv1
        var1 ∈ ℕ // comment
    @inv2
        var2 ⊆ ℕ

    theorem @inv3
        1 ∈ ℕ // multiline comment

variant
    var1 // comment

events
    event INITIALISATION // comment
      then
        @act1 var1 ≔ 1
        @act2 var2 ≔ {1}
    end

    convergent event evt1
      any
        prm1 // comment
      where
        @grd1 prm1 = 1
        @grd2 var1 ≠ prm1 // comment
        @grd3 var1 > 0
        theorem @grd4 var1 ≠ 0
      then
        @act1 var1 ≔ var1 − prm1 // comment
        @act2 var2 ≔ {1}
    end

end
